python/lsst/daf/butler/core/formatter.py

```diff
@@ -158,7 +159,34 @@ def toBytes(self, inMemoryDataset: Any) -> bytes:
         raise NotImplementedError("Type does not support writing to bytes.")
 
     @classmethod
-    @abstractmethod
+    def makeUpdatedLocation(cls, location: Location) -> Location:
+        """Return a new `Location` instance updated with this formatter's
+        extension.
+
+        Parameters
+        ----------
+        location : `Location`
+            The location to update.
+
+        Returns
+        -------
+        updated : `Location`
+            The updated location with a new file extension applied.
+
+        Raises
+        ------
+        NotImplementedError
+            Raised if there is no ``extension`` attribute associated with
+            this formatter.
+        """
+        location = copy.deepcopy(location)
+        try:
+            location.updateExtension(cls.extension)
+        except AttributeError:
+            raise NotImplementedError("No file extension registered with this formatter")
```

Comment:
Use `raise NotImplementedError(...) from None` here if you want to avoid "while handing this exception..." noise in the traceback.